### PR TITLE
Raise errors with a better name 

### PR DESCRIPTION
**Pull request by brunojabs**
Change the error raised in order to be more clear about of the use case where the error was raised

Card: https://trello.com/c/MGXY1qWM/3207-caze-raise-specific-error-for-each-use-case

### CODE REVIEW

File: lib/caze.rb [hunk 1]
```diff
@@ -59,5 +58,22 @@ def get_use_case_class(use_case_class)
 
       use_case_class
     end
+
+    # This method intends to raise the error inside the context of the use case
+    # So we can have a more specific
```
Comment (Lucasosf) [inline lib/caze.rb]: `So we can have a more specific error` ? oO

File: spec/caze_spec.rb [hunk 2]
```diff
@@ -114,10 +131,17 @@ module Dummy
 
     context 'using exception' do
       context 'when the use case raises an exception' do
-        it 'shows the use case name' do
+        it 'raises with the namespace of the use case' do
+          begin
+            namespaced_use_case.the_question
```
Comment (Lucasosf) [inline spec/caze_spec.rb]: I think you could do something like this:
```
expect {
  namespaced_use_case.the_question
}.to raise_error(DummyUseCaseTest::RuntimeError)
```

File: lib/caze.rb [hunk 3]
```diff
@@ -59,5 +58,22 @@ def get_use_case_class(use_case_class)
...
+    def raise_use_case_error(use_case, error)
+
```
Comment (philss) [inline lib/caze.rb]: Pls remove this empty line.

File: lib/caze.rb (hunk 1 above)
Comment (brunojabs) [inline lib/caze.rb]: 5e50503

File: spec/caze_spec.rb (hunk 2 above)
Comment (brunojabs) [inline spec/caze_spec.rb]: Yep, the 'DummyUseCaseTest::RuntimeError' does not exists on the context of the test. That's why we are doing this way

Comment (Lucasosf) [inline spec/caze_spec.rb]: Are sure? I tried and it's all right oO

Comment (brunojabs) [inline spec/caze_spec.rb]: 11d670a

File: lib/caze.rb (hunk 3 above)
Comment (brunojabs) [inline lib/caze.rb]: 5e50503

**Comment by Lucasosf** [comment]
Awesome, :shipit: :+1:

File: lib/caze.rb [hunk 4]
```diff
@@ -59,5 +58,21 @@ def get_use_case_class(use_case_class)
...
+      base_class = Class.new(error.class) do 
+        define_singleton_method :name do 
+          "#{use_case}::#{error.class}"
+        end
+      end
+                
+      error_class = use_case.const_set(demodulized_error_class, base_class)
+
+      raise error_class, error.message
```
Comment (philss) [inline lib/caze.rb]: I'm not sure if we are keeping the backtrace of the original exception. Is it possible to make sure that we are keeping it?

Comment (brunojabs) [inline lib/caze.rb]: We are not keeping, I tried to raise `raise error_class, error.message, error.backtrace` and `raise error_class, error.message, caller` and it does not work. But we'll create a card to fix later

**Comment by brunojabs** [comment]
@Lucasosf, @philss, @bezelga ping

**Comment by luizParreira** [comment]
:shipit:

File: lib/caze.rb (hunk 4 above)
Comment (philss) [inline lib/caze.rb]: Ok, we can improve that later.

**Comment by philss** [comment]
:shipit:

**Comment by philss** [comment]
@brunojabs don't forget to bump a new version.

**Comment by brunojabs** [comment]
@philss, @Lucasosf, @bezelga, @philss I've made a change to keep the backtrace, please check again, tks!

**Comment by bezelga** [comment]
:ship:

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,12 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
+## [0.2.0] - 2016-1-7
```
Comment (philss) [inline CHANGELOG.md]: Please use the format `YYYY-MM-DD`, as recommended by the [Keep a Changelog](http://keepachangelog.com/).

**Comment by philss** [comment]
:shipit:

**Pull request merged**